File: In Ruby/Pascal-Triangle.rb
def fact(n)
  (1..n).reduce(1, :*)
end

def ncr(n, r)
  fact(n) / (fact(r) * fact(n - r))
end

print "Enter number of rows: "
n = gets.to_i

(0...n).each do |i|
  print " " * (n - i)
  (0..i).each do |j|
    print "#{ncr(i, j)} "
  end
  puts
end

File: In Java/Pascal-Triangle.java
import java.util.Scanner;

public class PascalTriangle {
    static int fact(int n) {
        int f = 1;
        for (int i = 1; i <= n; i++) f *= i;
        return f;
    }

    static int ncr(int n, int r) {
        return fact(n) / (fact(r) * fact(n - r));
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.print("Enter number of rows: ");
        int n = sc.nextInt();

        for (int i = 0; i < n; i++) {
            for (int j = 0; j < n - i; j++) System.out.print(" ");
            for (int j = 0; j <= i; j++) {
                System.out.print(ncr(i, j) + " ");
            }
            System.out.println();
        }
        sc.close();
    }
}
